docs/doc-generator.py

```diff
@@ -16,6 +16,56 @@ def gen_id(string):
     return string.split("(")[0].replace(".", "_")
 
 
+def pretty_function_name(name):
+    outer_match = re.search(r"(\w+)\((.*)\)", name)
+    # If this is not a function, just return the raw name
+    if outer_match is not None:
+        fn_name = outer_match.group(1)
+        arg_string = outer_match.group(2)
+
+        args = []
+        if arg_string:
+            # Try to build a list of arguments
+            comma_separated = arg_string.split(',')
+            # Spliting on , has side effects when the default value has , in it. We'll try
+            # to rectify that by re-joining string which don't look like identifiers
+            for part in comma_separated:
+                # If this is a new argument
+                arg_regex = re.search(r"([A-z]+)(=?)(.*)", part)
```

Comment:
If I understand this correctly, the regex looks for parameters and potential default values. If so, you want something like this:
```suggestion
                # If this is a new argument
                arg_regex = re.search(r"([A-z_][0-9A-z_]*)(=?)(.*)", part)
```